File: MiniDOMWebLibs1/MiniDOM/IPInfo/CIPInfo.vb
Imports minidom
Imports minidom.Databases
Imports minidom.Sistema

Partial Class WebSite

    Public Class CIPInfo
        Inherits DBObjectBase

        Private m_IP As String
        Private m_Descrizione As String

        Public Sub New()
            Me.m_IP = ""
            Me.m_Descrizione = ""
        End Sub

        Public Property IP As String
            Get
                Return Me.m_IP
            End Get
            Set(value As String)
                value = Trim(value)
                Dim oldValue As String = Me.m_IP
                If (oldValue = value) Then Exit Property
                Me.m_IP = value
                Me.DoChanged("IP", value, oldValue)
            End Set
        End Property

        Public Property Descrizione As String
            Get
                Return Me.m_Descrizione
            End Get
            Set(value As String)
                value = Trim(value)
                Dim oldValue As String = Me.m_Descrizione
                If (oldValue = value) Then Exit Property
                Me.m_Descrizione = value
                Me.DoChanged("Descrizione", value, oldValue)
            End Set
        End Property


        Protected Overrides Function GetConnection() As CDBConnection
            Return LOGConn
        End Function

        Public Overrides Function GetModule() As CModule
            Return IPInfo.Module
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_IPInfo"
        End Function

        Public Overrides Function ToString() As String
            Return "(" & Me.m_IP & ", " & Me.m_Descrizione & ")"
        End Function

        Protected Overrides Function LoadFromRecordset(reader As DBReader) As Boolean
            Me.m_IP = reader.Read("IP", Me.m_IP)
            Me.m_Descrizione = reader.Read("Descrizione", Me.m_Descrizione)
            Return MyBase.LoadFromRecordset(reader)
        End Function

        Protected Overrides Function SaveToRecordset(writer As DBWriter) As Boolean
            writer.Write("IP", Me.m_IP)
            writer.Write("Descrizione", Me.m_Descrizione)
            Return MyBase.SaveToRecordset(writer)
        End Function

        Protected Overrides Sub XMLSerialize(writer As XML.XMLWriter)
            writer.WriteAttribute("IP", Me.m_IP)
            MyBase.XMLSerialize(writer)
            writer.WriteTag("Descrizione", Me.m_Descrizione)
        End Sub

        Protected Overrides Sub SetFieldInternal(fieldName As String, fieldValue As Object)
            Select Case fieldName
                Case "IP" : Me.m_IP = XML.Utils.Serializer.DeserializeString(fieldValue)
                Case "Descrizione" : Me.m_Descrizione = XML.Utils.Serializer.DeserializeString(fieldValue)
                Case Else : MyBase.SetFieldInternal(fieldName, fieldValue)
            End Select
        End Sub


    End Class

End Class

File: MiniDOMWebLibs1/MiniDOM/IPInfo/CIPInfo.cs
using System.Globalization;
using DMD;

namespace minidom
{
    public partial class WebSite
    {
        public class CIPInfo : Databases.DBObjectBase
        {
            private string m_IP;
            private string m_Descrizione;

            public CIPInfo()
            {
                m_IP = "";
                m_Descrizione = "";
            }

            public string IP
            {
                get
                {
                    return m_IP;
                }

                set
                {
                    value = Strings.Trim(value);
                    string oldValue = m_IP;
                    if (CultureInfo.CurrentCulture.CompareInfo.Compare(oldValue ?? "", value ?? "", CompareOptions.IgnoreCase | CompareOptions.IgnoreKanaType | CompareOptions.IgnoreWidth) == 0)
                        return;
                    m_IP = value;
                    DoChanged("IP", value, oldValue);
                }
            }

            public string Descrizione
            {
                get
                {
                    return m_Descrizione;
                }

                set
                {
                    value = Strings.Trim(value);
                    string oldValue = m_Descrizione;
                    if (CultureInfo.CurrentCulture.CompareInfo.Compare(oldValue ?? "", value ?? "", CompareOptions.IgnoreCase | CompareOptions.IgnoreKanaType | CompareOptions.IgnoreWidth) == 0)
                        return;
                    m_Descrizione = value;
                    DoChanged("Descrizione", value, oldValue);
                }
            }

            protected override Databases.CDBConnection GetConnection()
            {
                return Databases.LOGConn;
            }

            public override Sistema.CModule GetModule()
            {
                return IPInfo.Module;
            }

            public override string GetTableName()
            {
                return "tbl_IPInfo";
            }

            public override string ToString()
            {
                return "(" + m_IP + ", " + m_Descrizione + ")";
            }

            protected override bool LoadFromRecordset(Databases.DBReader reader)
            {
                m_IP = reader.Read("IP", ref m_IP);
                m_Descrizione = reader.Read("Descrizione", ref m_Descrizione);
                return base.LoadFromRecordset(reader);
            }

            protected override bool SaveToRecordset(Databases.DBWriter writer)
            {
                writer.Write("IP", m_IP);
                writer.Write("Descrizione", m_Descrizione);
                return base.SaveToRecordset(writer);
            }

            protected override void XMLSerialize(XML.XMLWriter writer)
            {
                writer.WriteAttribute("IP", m_IP);
                base.XMLSerialize(writer);
                writer.WriteTag("Descrizione", m_Descrizione);
            }

            protected override void SetFieldInternal(string fieldName, object fieldValue)
            {
                switch (fieldName ?? "")
                {
                    case var @case when CultureInfo.CurrentCulture.CompareInfo.Compare(@case, "IP", CompareOptions.IgnoreCase | CompareOptions.IgnoreKanaType | CompareOptions.IgnoreWidth) == 0:
                        {
                            m_IP = DMD.XML.Utils.Serializer.DeserializeString(fieldValue);
                            break;
                        }

                    case var case1 when CultureInfo.CurrentCulture.CompareInfo.Compare(case1, "Descrizione", CompareOptions.IgnoreCase | CompareOptions.IgnoreKanaType | CompareOptions.IgnoreWidth) == 0:
                        {
                            m_Descrizione = DMD.XML.Utils.Serializer.DeserializeString(fieldValue);
                            break;
                        }

                    default:
                        {
                            base.SetFieldInternal(fieldName, fieldValue);
                            break;
                        }
                }
            }
        }
    }
}